File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.length()<2){
            return s.length();
        }
        HashMap <Character, Integer> map = new HashMap<>();
        int maxi = 0;
        int l = 0;
        for(int i=0;i<s.length();i++){
            if (map.containsKey(s.charAt(i))){
                l = Math.max(l,map.get(s.charAt(i))+1);
                

            }
            maxi = Math.max(maxi, i-l+1);

            map.put(s.charAt(i),i);

        }

        return maxi;
        
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxi = 0
        l = 0
        mapp = {}
        for i in range(len(s)):
            if s[i] in mapp:
                l = max(l,mapp[s[i]]+1)
            
            mapp[s[i]] = i

            length= i-l+1
            maxi = max(length,maxi)
        return maxi

            

            
            
        
        